pkg/standalone-utils/contracts/relayer/GaugeActions.sol

```diff
@@ -96,4 +106,47 @@ abstract contract GaugeActions is IBaseRelayerLibrary {
             gauges[i].claim_rewards(msg.sender);
         }
     }
+
+    /**
+     * @notice Perform a user checkpoint for the given user at the given set of gauges.
+     * @dev Both mainnet and child chain gauges are supported.
+     */
+    function gaugeCheckpoint(address user, IStakingLiquidityGauge[] calldata gauges) external payable {
+        if (_isL2Relayer) {
+            _checkpointL2Gauges(user, gauges);
+        } else {
+            _checkpointL1Gauges(user, gauges);
+        }
+    }
+
+    function _checkpointL2Gauges(address user, IStakingLiquidityGauge[] calldata gauges) internal {
+        uint256 numGauges = gauges.length;
+        // In L2s (child chain gauges), `user_checkpoint` is not permissioned, so we can just call it directly.
+        for (uint256 i = 0; i < numGauges; ++i) {
+            gauges[i].user_checkpoint(user);
+        }
+    }
+
+    function _checkpointL1Gauges(address user, IStakingLiquidityGauge[] calldata gauges) internal {
+        uint256 numGauges = gauges.length;
+        IVault.UserBalanceOp[] memory ops = new IVault.UserBalanceOp[](numGauges);
+
+        // In mainnet, `user_checkpoint` is permissioned for liquidity gauges, so we cannot call it directly.
+        // The checkpoint in this case is triggered by a gauge token transfer from the user to itself.
+        // This is done using the Vault's allowance which is unlimited for gauge tokens.
```

Comment:
```suggestion
        // However, some non-permissioned actions cause the gauge to checkpoint a user as a side effect,
        // even if the operation itself is a no-op. The simplest of these is a gauge token transfer, which we
        // perform here. Since the Vault has an unlimited allowance for gauge tokens, and user balance
        // operations use the Vault allowance, no approvals are necessary.
```
Maybe make this "Curve quirk" a little clearer? (Also a Vault user balance trick.)